core/src/PrognosticData.cpp

```diff
@@ -51,6 +58,20 @@ void PrognosticData::configure()
     tryConfigure(pDynamics);
 
     tryConfigure(iceGrowth);
+
+    boolCheckAll() = Configured::getConfiguration(keyMap.at(CHECKFIELDS_KEY), checkFieldsDefault);
+    boolCheckFields
+        = Configured::getConfiguration(keyMap.at(CHECKFIELDS_KEY), checkFieldsFastDefault);
+    if (boolCheckAll()) {
+        for (const auto& field : getStore().getAllData()) {
+            fieldsToCheck.emplace_back(field.first, field.second);
+        }
+    } else if (boolCheckFields) {
+        fieldsToCheck.emplace_back("thickness", &hiceAdvection);
+        fieldsToCheck.emplace_back("concentration", &ciceAdvection);
+        fieldsToCheck.emplace_back("u_velocity", getStore().getArrayRef(Protected::ICE_U));
+        fieldsToCheck.emplace_back("v_velocity", getStore().getArrayRef(Protected::ICE_V));
```

Comment:
Consider adding a function to the superclass that takes a `std::map<std::string, ModelArrayReference>` argument, rather than repeating `emplace_back` so many times in the derived class code.

Reply:
Nice suggestion!